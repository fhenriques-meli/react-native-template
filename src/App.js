import React, { useEffect } from 'react';
import SplashScreen from 'react-native-splash-screen';
import { Provider } from 'react-redux';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { PersistGate } from 'redux-persist/integration/react';
import store, { persistor } from '~/config/store';

import { Home } from '~/screens';

const Stack = createStackNavigator();

const App = () => {
  useEffect(() => {
    SplashScreen.hide();
  }, []);

  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <NavigationContainer>
          <Stack.Navigator headerMode="none">
            <Stack.Screen name="Home" component={Home} headerMode="none" />
            {/* <Stack.Screen name="Color" component={Color} headerMode="none" /> */}
          </Stack.Navigator>
        </NavigationContainer>
      </PersistGate>
    </Provider>
  );
};

export default App;
