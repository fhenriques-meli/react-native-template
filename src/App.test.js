import React from 'react';
import renderer from 'react-test-renderer';

import App from './App';

jest.useFakeTimers();

describe('<App />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.type).toBe('RNCSafeAreaView');
  });

  it('renders correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
