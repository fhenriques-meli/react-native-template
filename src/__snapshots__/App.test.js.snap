// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "opacity": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
              ]
            }
          >
            <RNCSafeAreaView
              onInsetsChange={[Function]}
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "paddingBottom": 0,
                      "paddingLeft": 0,
                      "paddingRight": 0,
                      "paddingTop": 20,
                    },
                    Array [
                      Object {
                        "backgroundColor": "#fff",
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                      },
                    ],
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "borderBottomColor": "#bbb",
                        "borderBottomWidth": 0.5,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#ffffff",
                        "borderRadius": 4,
                        "elevation": 0,
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      accessibilityRole="button"
                      accessibilityStates={Array []}
                      accessible={true}
                      focusable={false}
                      hitSlop={
                        Object {
                          "bottom": 6,
                          "left": 6,
                          "right": 6,
                          "top": 6,
                        }
                      }
                      isTVSelectable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Array [
                            Object {
                              "alignItems": "center",
                              "justifyContent": "center",
                              "margin": 6,
                              "overflow": "hidden",
                            },
                            Object {
                              "borderRadius": 18,
                              "height": 36,
                              "width": 36,
                            },
                            undefined,
                            undefined,
                          ],
                        ]
                      }
                    >
                      <View
                        style={null}
                      >
                        <Text />
                      </View>
                    </View>
                    <TextInput
                      accessibilityRole="search"
                      accessibilityTraits="search"
                      allowFontScaling={true}
                      keyboardAppearance="light"
                      onChangeText={[Function]}
                      placeholder="E.g. #love #tbt #me"
                      placeholderTextColor="rgba(0, 0, 0, 0.54)"
                      rejectResponderTermination={true}
                      returnKeyType="search"
                      selectionColor="#6200ee"
                      style={
                        Array [
                          Object {
                            "alignSelf": "stretch",
                            "flex": 1,
                            "fontSize": 18,
                            "minWidth": 0,
                            "paddingLeft": 8,
                            "textAlign": "left",
                          },
                          Object {
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          undefined,
                        ]
                      }
                      underlineColorAndroid="transparent"
                      value={null}
                    />
                    <View
                      accessibilityRole="button"
                      accessibilityStates={
                        Array [
                          "disabled",
                        ]
                      }
                      accessible={true}
                      focusable={true}
                      hitSlop={
                        Object {
                          "bottom": 6,
                          "left": 6,
                          "right": 6,
                          "top": 6,
                        }
                      }
                      isTVSelectable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Array [
                            Object {
                              "alignItems": "center",
                              "justifyContent": "center",
                              "margin": 6,
                              "overflow": "hidden",
                            },
                            Object {
                              "borderRadius": 18,
                              "height": 36,
                              "width": 36,
                            },
                            Object {
                              "opacity": 0.32,
                            },
                            undefined,
                          ],
                        ]
                      }
                    >
                      <View
                        style={null}
                      >
                        <Text />
                      </View>
                    </View>
                  </View>
                </View>
                <RCTScrollView
                  keyboardShouldPersistTaps="handled"
                  style={
                    Array [
                      Object {
                        "flexBasis": 0,
                        "flexGrow": 1,
                        "flexShrink": 1,
                      },
                    ]
                  }
                >
                  <View>
                    <View
                      style={
                        Array [
                          Object {
                            "marginVertical": 8,
                          },
                          Array [
                            Object {
                              "paddingBottom": 72,
                            },
                          ],
                        ]
                      }
                      theme={
                        Object {
                          "animation": Object {
                            "scale": 1,
                          },
                          "colors": Object {
                            "accent": "#03dac4",
                            "backdrop": "rgba(0, 0, 0, 0.5)",
                            "background": "#f6f6f6",
                            "disabled": "rgba(0, 0, 0, 0.26)",
                            "error": "#B00020",
                            "notification": "#f50057",
                            "onBackground": "#000000",
                            "onSurface": "#000000",
                            "placeholder": "rgba(0, 0, 0, 0.54)",
                            "primary": "#6200ee",
                            "surface": "#ffffff",
                            "text": "#000000",
                          },
                          "dark": false,
                          "fonts": Object {
                            "light": Object {
                              "fontFamily": "System",
                              "fontWeight": "300",
                            },
                            "medium": Object {
                              "fontFamily": "System",
                              "fontWeight": "500",
                            },
                            "regular": Object {
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            "thin": Object {
                              "fontFamily": "System",
                              "fontWeight": "100",
                            },
                          },
                          "roundness": 4,
                        }
                      }
                    >
                      <Text
                        numberOfLines={1}
                        style={
                          Array [
                            Object {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                              "textAlign": "left",
                            },
                            Array [
                              Object {
                                "paddingHorizontal": 16,
                                "paddingVertical": 13,
                              },
                              Object {
                                "color": "rgba(0, 0, 0, 0.54)",
                                "fontFamily": "System",
                                "fontWeight": "500",
                              },
                              undefined,
                            ],
                          ]
                        }
                      />
                    </View>
                  </View>
                </RCTScrollView>
                <View
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "flexBasis": 0,
                        "flexDirection": "row",
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "justifyContent": "space-between",
                        "left": 0,
                        "marginBottom": 16,
                        "marginLeft": 16,
                        "marginRight": 16,
                        "marginTop": 16,
                        "position": "absolute",
                        "right": 0,
                      },
                    ]
                  }
                >
                  <View
                    pointerEvents="none"
                    style={
                      Object {
                        "backgroundColor": "#03dac4",
                        "borderRadius": 28,
                        "elevation": 6,
                        "opacity": 0,
                        "shadowColor": "#000000",
                        "shadowOffset": Object {
                          "height": 5,
                          "width": 0,
                        },
                        "shadowOpacity": 0.24,
                        "shadowRadius": 6,
                        "transform": Array [
                          Object {
                            "scale": 0,
                          },
                        ],
                      }
                    }
                  >
                    <View
                      accessibilityLabel="SHUFFLE"
                      accessibilityRole="button"
                      accessibilityStates={Array []}
                      accessible={true}
                      focusable={true}
                      isTVSelectable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Object {
                            "borderRadius": 28,
                          },
                        ]
                      }
                    >
                      <View
                        pointerEvents="none"
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "center",
                            },
                            Object {
                              "height": 48,
                              "paddingHorizontal": 16,
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "justifyContent": "center",
                              },
                              Object {
                                "height": 24,
                                "width": 24,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "opacity": 1,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                                "transform": Array [
                                  Object {
                                    "rotate": "0deg",
                                  },
                                ],
                              }
                            }
                          >
                            <Text />
                          </View>
                        </View>
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000000",
                                "fontFamily": "System",
                                "fontWeight": "400",
                                "textAlign": "left",
                              },
                              Array [
                                Object {
                                  "marginHorizontal": 8,
                                },
                                Object {
                                  "color": "rgba(0, 0, 0, .54)",
                                  "fontFamily": "System",
                                  "fontWeight": "500",
                                },
                              ],
                            ]
                          }
                        >
                          SHUFFLE
                        </Text>
                      </View>
                    </View>
                  </View>
                  <View
                    pointerEvents="none"
                    style={
                      Object {
                        "backgroundColor": "rgba(0, 0, 0, 0.12)",
                        "borderRadius": 28,
                        "elevation": 0,
                        "opacity": 0,
                        "transform": Array [
                          Object {
                            "scale": 0,
                          },
                        ],
                      }
                    }
                  >
                    <View
                      accessibilityLabel="COPY (0)"
                      accessibilityRole="button"
                      accessibilityStates={
                        Array [
                          "disabled",
                        ]
                      }
                      accessible={true}
                      focusable={true}
                      isTVSelectable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Object {
                            "borderRadius": 28,
                          },
                        ]
                      }
                    >
                      <View
                        pointerEvents="none"
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "justifyContent": "center",
                            },
                            Object {
                              "height": 48,
                              "paddingHorizontal": 16,
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "justifyContent": "center",
                              },
                              Object {
                                "height": 24,
                                "width": 24,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "opacity": 1,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                                "transform": Array [
                                  Object {
                                    "rotate": "0deg",
                                  },
                                ],
                              }
                            }
                          >
                            <Text />
                          </View>
                        </View>
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#000000",
                                "fontFamily": "System",
                                "fontWeight": "400",
                                "textAlign": "left",
                              },
                              Array [
                                Object {
                                  "marginHorizontal": 8,
                                },
                                Object {
                                  "color": "rgba(0, 0, 0, 0.32)",
                                  "fontFamily": "System",
                                  "fontWeight": "500",
                                },
                              ],
                            ]
                          }
                        >
                          COPY (0)
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </RNCSafeAreaView>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="auto"
      style={
        Object {
          "backgroundColor": "rgb(255, 255, 255)",
          "borderTopColor": "rgb(199, 199, 204)",
          "borderTopWidth": 0.5,
          "bottom": 0,
          "elevation": 8,
          "height": 50,
          "left": 0,
          "paddingBottom": 0,
          "right": 0,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityLabel="Home1, tab, 1 of 4"
          accessibilityRole="button"
          accessibilityStates={
            Array [
              "selected",
            ]
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <Text />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <Text />
            </View>
          </View>
        </View>
        <View
          accessibilityLabel="Home2, tab, 2 of 4"
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <Text />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <Text />
            </View>
          </View>
        </View>
        <View
          accessibilityLabel="Home3, tab, 3 of 4"
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <Text />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <Text />
            </View>
          </View>
        </View>
        <View
          accessibilityLabel="Home4, tab, 4 of 4"
          accessibilityRole="button"
          accessibilityStates={Array []}
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 0,
                  },
                ]
              }
            >
              <Text />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": "100%",
                    "justifyContent": "center",
                    "minWidth": 25,
                    "position": "absolute",
                    "width": "100%",
                  },
                  Object {
                    "opacity": 1,
                  },
                ]
              }
            >
              <Text />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
